### ISSUE TITLE
AI karts not present in lobby or race on Windows with 1.1-rc1

### ISSUE THREAD

**Issue by danielnelson** [supertuxkart/stk-code#4184]
## Description

With 1.1-rc1, when creating a LAN server on Windows, AI karts do not join lobby nor are they present in the race.  On Linux, AI karts are working with the same setup.

## Steps to reproduce

- Online -> Local Networking -> Create Server
- Set max players to 8
- Set number of AI karts to 6
- Click create

**Expected**
6 AI karts will be shown in the lobby

**Actual**
No AI karts join

## Configuration

STK release version: `SuperTuxKart 1.1.0-rc1 installer-64bit.exe`

STK Source (PPA, distribution package, official binary, etc.): official binary

System: Windows 7 64-bit

Graphics card: Radeon RX 580

CPU: Ryzen 1700

Gamepads/keyboards models if related to the issue: N/A

## Additional information

stdout.log: [stdout.log](https://github.com/supertuxkart/stk-code/files/4004063/stdout.log)

server.log: [server.log](https://github.com/supertuxkart/stk-code/files/4004067/server.log)

input.xml: N/A

Error message: N/A

**Comment by Benau** [comment]
what is server_ai.log ?

**Comment by danielnelson** [comment]
I didn't have a `server_ai.log`.  I ran the command shown in the `server.log` while in listening in the lobby and it appeared to work: the AI karts join the lobby and the logfile is created.

**Comment by Benau** [comment]
Seems the your windows failed to createProcess for child of child process... (antivirus problems?)

You need to start ai instance manually as a workaround for now ![](https://lihkg.com/assets/faces/pigxm/upset_r.gif)

How about other stk windows fans? ![](https://lihkg.com/assets/faces/pigxm/dance.gif)

**Comment by danielnelson** [comment]
Thanks, I'll use the workaround for now.  I don't use an antivirus but let me know if there is anything else I can try to help debug the issue.  Maybe there is a way to get at the error code from createProcess, something like strace for Windows?

**Comment by Benau** [comment]
[supertuxkart.zip](https://github.com/supertuxkart/stk-code/files/4006110/supertuxkart.zip)
Try this exe inside? (replace C:\games\stk\SuperTuxKart 1.1.0-rc1\supertuxkart.exe) ![](https://lihkg.com/assets/faces/pigxm/dance2.gif)

**Comment by danielnelson** [comment]
That seems to have fixed it, all the AI karts now join up without issues:
```
Fri Dec 27 12:36:40 2019 [info   ] STKHost: Host initialized.
Fri Dec 27 12:36:40 2019 [info   ] STKHost: Server port is 2759
Fri Dec 27 12:36:40 2019 [info   ] ServerLobby: Resetting the server to its initial state.
Fri Dec 27 12:36:40 2019 [info   ] ProtocolManager: A 11ServerLobby protocol has been started.
Fri Dec 27 12:36:40 2019 [info   ] main: Creating a LAN server 'dbn's&#x20;server'.
Fri Dec 27 12:36:40 2019 [info   ] STKHost: Listening has been started.
Fri Dec 27 12:36:40 2019 [warn   ] LayoutManager: Widget 'timeout-message' has a height of 0 (left_space = 0, fraction = 1.000000, max_width = )

Fri Dec 27 12:36:40 2019 [info   ] STKHost: 127.0.0.1:55686 has just connected. There are now 1 peers.
Fri Dec 27 12:36:40 2019 [info   ] ServerLobby: Message of type 1 received.
Fri Dec 27 12:36:40 2019 [info   ] ServerLobby: New player dbn with online id 0 from 127.0.0.1:55686 with SuperTuxKart/1.1-rc1 (Windows).
Fri Dec 27 12:36:41 2019 [info   ] STKHost: 127.0.0.1:55688 has just connected. There are now 2 peers.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: Message of type 1 received.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot 1 with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot 2 with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot 3 with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot 4 with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:41 2019 [info   ] ServerLobby: New player Bot 5 with online id 0 from 127.0.0.1:55688 with AI.
Fri Dec 27 12:36:50 2019 [info   ] ServerLobby: Message of type 5 received.
Fri Dec 27 12:36:52 2019 [info   ] ServerLobby: Message of type 7 received.
Fri Dec 27 12:36:53 2019 [info   ] ServerLobby: Message of type 16 received.
```

**Comment by Benau** [comment]
Fix in 10c0515c88e073659bfda9b98e712376fc67a008 ![](https://lihkg.com/assets/faces/pigxm/clap.gif)
